arlas-core/src/main/java/io/arlas/server/utils/RequestFieldsExtractor.java

```diff
@@ -0,0 +1,165 @@
+        if (Arrays.asList(Request.class, Count.class).contains(request.getClass())) {
+            requestExtractor = new BasicRequestFieldsExtractor();
+        } else if (request.getClass() == Search.class) {
+            requestExtractor = new SearchRequestFieldsExtractor();
+        } else if (request.getClass() == AggregationsRequest.class) {
+            requestExtractor = new AggregationRequestFieldsExtractor();
+        } else if (request.getClass() == RangeRequest.class) {
+            requestExtractor = new RangeRequestFieldsExtractor();
+        } else {
+            throw new InternalServerErrorException("Request type is not supported");
+        }
+
+        return requestExtractor.getCols(request);
+    }
+
+    private interface IRequestFieldsExtractor<T extends Request> {
+        Stream<String> getCols(T request);
+    }
+
+    private static class BasicRequestFieldsExtractor<T extends Request> implements IRequestFieldsExtractor<T> {
+
+        public Stream<String> getCols(T request) {
+            return Stream.concat(getFCols(request), getQCols(request));
+        }
+
+        private Stream<String> getFCols(Request request) {
+            return Optional.ofNullable(request.filter).flatMap(filter -> Optional.ofNullable(filter.f))
+                    .map(f -> f.stream()
+                            .flatMap(fList ->
+                                    fList.stream().map(fFilter -> fFilter.field))
+                    ).orElse(Stream.of());
+        }
+
+        private Stream<String> getQCols(Request request) {
+            return  Optional.ofNullable(request.filter).flatMap(filter -> Optional.ofNullable(filter.q)
+                    .map(q ->
+                            q.stream().flatMap(qList ->
+                                    qList.stream()
+                                            .filter(qFilter -> qFilter.contains(":"))
+                                            .map(qFilter -> qFilter.split(":")[0]))
+                    )).orElse(Stream.of());
+        }
+    }
+
+    private static class SearchRequestFieldsExtractor extends BasicRequestFieldsExtractor<Search> {
+
+        @Override
+        public Stream<String> getCols(Search request) {
+            return Stream.of(
+                    super.getCols(request),
+                    getSortCols(request),
+                    getIncludeCols(request),
+                    getExcludeCols(request)
+                    )
+                    .flatMap(x -> x);
+        }
+
+        private Stream<String> getSortCols(Search search) {
+            return Optional.ofNullable(search.page).flatMap(p -> Optional.ofNullable(p.sort)
+                    .map(sort -> Arrays.stream(sort.split(",")).map(sortCol -> {
+                        if (sortCol.startsWith("-")) {
+                            return sortCol.substring(1);
+                        } else if (sortCol.contains(":")) {
```

Comment:
After thinking twice to this I should we should rather not consider `geodistance` by extracting the sort fields, because `geodistance` isn't really a field. Are you OK with it?